packages/loot-design/src/components/modals/SelectLinkedAccounts.js

```diff
@@ -1,254 +1,200 @@
+                  <TableRow
+                    externalAccount={item}
+                    linkedAccount={localAccounts.find(
+                      acc => acc.account_id === item.account_id,
+                    )}
+                    chosenAccount={
+                      chosenAccounts[item.account_id] === addAccountOption.value
+                        ? {
+                            id: addAccountOption.value,
+                            name: addAccountOption.label,
+                          }
+                        : localAccounts.find(
+                            acc => chosenAccounts[item.account_id] === acc.id,
+                          )
+                    }
+                    unlinkedAccounts={unlinkedAccounts}
+                    onSetLinkedAccount={onSetLinkedAccount}
+                  />
                 </View>
               )}
-            </View>
+            />
           </View>
-
           <View
             style={{
               flexDirection: 'row',
               justifyContent: 'flex-end',
-              marginTop: 30,
+              marginTop: 10,
             }}
           >
-            <Button style={{ marginRight: 10 }} onClick={modalProps.onClose}>
-              Cancel
-            </Button>
-            <Button primary onClick={onNext} disabled={!chosenAccounts.length}>
+            <Button
+              primary
+              onClick={onNext}
+              disabled={!Object.keys(chosenAccounts).length}
+            >
               Link accounts
             </Button>
           </View>
-        </View>
+        </>
       )}
     </Modal>
   );
 }
+
+function TableRow({
+  externalAccount,
+  linkedAccount,
+  chosenAccount,
+  unlinkedAccounts,
+  onSetLinkedAccount,
+}) {
+  const [focusedField, setFocusedField] = useState(null);
+
+  const chosenAccountOption = chosenAccount && {
+    value: chosenAccount.id,
+    label: chosenAccount.name,
+  };
+
+  const availableAccountOptions = [
+    ...unlinkedAccounts.map(acct => ({
+      value: acct.id,
+      label: acct.name,
+    })),
+    chosenAccount?.id !== addAccountOption.value && chosenAccountOption,
+    addAccountOption,
+  ].filter(Boolean);
+
+  return (
+    <Row style={{ backgroundColor: 'white' }}>
+      <Field width={200}>{externalAccount.name}</Field>
+      <Field
+        width="flex"
+        truncate={focusedField !== 'account'}
+        onClick={() => !linkedAccount && setFocusedField('account')}
+      >
+        {focusedField === 'account' ? (
+          <Autocomplete
+            autoFocus
+            options={availableAccountOptions}
+            onSelect={value => {
+              onSetLinkedAccount(externalAccount, value);
+            }}
+            onBlur={() => setFocusedField(null)}
+            value={chosenAccountOption}
+          />
+        ) : (
+          linkedAccount?.name || chosenAccount?.name
+        )}
+      </Field>
+      <Field width="flex">
+        {!linkedAccount &&
```

Comment:
Not a fan of negative conditionals, could this easily be flipped or no?

Reply:
No.

If there IS a linked account (existing before the modal was opened): show nothing in the column

If there is NO linked account: depending on the local selection (chosenAccount) - either show "add" or "remove" button